presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPruneMarkDistinctColumns.java

```diff
@@ -0,0 +1,116 @@
+import com.facebook.presto.sql.planner.plan.Assignments;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.markDistinct;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.strictProject;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.values;
+
+public class TestPruneMarkDistinctColumns
+        extends BaseRuleTest
+{
+    @Test
+    public void testMarkerSymbolNotReferenced()
+            throws Exception
+    {
+        tester().assertThat(new PruneMarkDistinctColumns())
+                .on(p ->
+                {
+                    Symbol key = p.symbol("key", BIGINT);
+                    Symbol key2 = p.symbol("key2", BIGINT);
+                    Symbol mark = p.symbol("mark", BIGINT);
+                    Symbol unused = p.symbol("unused", BIGINT);
+                    return p.project(
+                            Assignments.of(key2, key.toSymbolReference()),
+                            p.markDistinct(mark, ImmutableList.of(key), Optional.empty(), p.values(key, unused)));
+                })
+                .matches(
+                        strictProject(
+                                ImmutableMap.of("key2", PlanMatchPattern.expression("key")),
+                                values(ImmutableList.of("key", "unused"))));
+    }
+
+    @Test
+    public void testSourceSymbolNotReferenced()
+            throws Exception
+    {
+        tester().assertThat(new PruneMarkDistinctColumns())
+                .on(p ->
+                {
+                    Symbol key = p.symbol("key", BIGINT);
+                    Symbol mark = p.symbol("mark", BIGINT);
+                    Symbol hash = p.symbol("hash", BIGINT);
+                    Symbol unused = p.symbol("unused", BIGINT);
+                    return p.project(
+                            Assignments.identity(mark),
+                            p.markDistinct(
+                                    mark,
+                                    ImmutableList.of(key),
+                                    Optional.of(hash),
+                                    p.values(key, hash, unused)));
+                })
+                .matches(
+                        strictProject(
+                                ImmutableMap.of("mark", PlanMatchPattern.expression("mark")),
+                                markDistinct("mark", ImmutableList.of("key"), Optional.of("hash"),
+                                        strictProject(
+                                                ImmutableMap.of(
+                                                        "key", PlanMatchPattern.expression("key"),
+                                                        "hash", PlanMatchPattern.expression("hash")),
```

Comment:
Yes, I'll do that, thanks.  The qualification was left over from a different test where PlanBuilder.expression was being imported.